src/main/java/bio/terra/workspace/app/controller/ControlledGoogleResourceApiController.java

```diff
@@ -73,7 +82,12 @@ public ControlledGoogleResourceApiController(
 
   @Override
   public ResponseEntity<GoogleBucketStoredAttributes> getBucket(UUID id, UUID resourceId) {
-    return new ResponseEntity<>(controlledResourceService.getBucket(id, resourceId), HttpStatus.OK);
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    ControlledResource controlledResource =
+        controlledResourceService.getControlledResource(id, resourceId, userRequest);
+    GoogleBucketStoredAttributes response =
+        controlledResource.castToGcsBucketResource().toApiModel();
```

Comment:
I'm open to suggestions. We could cast directly. I went with the belt'n'suspender of having common checking of the resource type along with the cast.

Another alternative would be for each specialized class to implement a static `fromControlledResource` so this would be `ControlledGcsBucketResource.fromControlledResource(controlledResource).toApiModel();`

It couldn't be an abstract method, but could be in all of the specializations by convention.

Reply:
I'll leave it as is for now. It is not hard to switch later if we want to.